addons/sale_timesheet/models/sale_order.py

```diff
@@ -78,6 +79,29 @@ class SaleOrderLine(models.Model):
 
     qty_delivered_method = fields.Selection(selection_add=[('timesheet', 'Timesheets')])
     analytic_line_ids = fields.One2many(domain=[('project_id', '=', False)])  # only analytic lines, not timesheets (since this field determine if SO line came from expense)
+    delivery_info = fields.Char(compute='_compute_delivery_info')
+
+    def name_get(self):
+        res = super(SaleOrderLine, self).name_get()
+        if self.env.context.get('with_remaining_hours'):
+            for index, name_get in enumerate(res):
+                id , name = name_get
+                sol = self.filtered(lambda sol: sol.id == id)
+                if sol.product_id and sol.product_id.service_policy == 'ordered_timesheet':
+                    name = "%(name)s - %(delivery_info)s" % {
+                        'name': name,
+                        'delivery_info': sol.delivery_info
+                    }
+                res[index] = (id, name)
+        return res
+
+    @api.depends('qty_delivered', 'product_uom_qty')
+    def _compute_delivery_info(self):
+        for sol in self:
+            sol.delivery_info = _('Delivered %g / %g') % (
+                float_round(sol.qty_delivered, precision_digits=2) or 0.0,
```

Comment:
Use rounding of uom
`fields.Float.round(sol.qty_delivered, precision_rounding = sol.product_uom_id.rounding)`
and use format_float, to format float in user lang.
https://github.com/odoo/odoo/pull/54407/files#diff-61162ac65633a1c7b054fc83ce1813f1a7984e3169ff36021713ef441f62a208R1488

Reply:
👍  I'll use Float.round. It uses the same behind but this is indeed cleaner!
I will not use the rouding of uom even if it is not ideal as for this provides crazy numbers for hours (which is the main purpose of this task).